### ISSUE TITLE
Or validation is not working as intended

### ISSUE THREAD

**Issue by karantan** [crdoconnor/strictyaml#31]
When you load yaml file and your scheme has an or validator you cannot change the type of the data afterwards.

Example:

```python
from strictyaml import Map, Bool, Int, YAMLValidationError, load

schema = Map({"a": Bool() | Int()})
data = load('a: yes', schema)
data['a'] = 5
*** strictyaml.exceptions.YAMLValidationError: when expecting a boolean value (one of "yes", "true", "on", "1", "y", "no", "false", "off", "0", "n")
found an arbitrary integer
  in "None", line 1, column 1:
    '5'
     ^ (line: 1)
```

I found a workaround. You actually need to do is change the validator:

```python
from strictyaml import Map, Bool, Int, YAMLValidationError, load

schema = Map({"a": Bool() | Int()})
data = load('a: yes', schema)
data['a']._validator = Int()
data['a'] = 5
assert data['a'] == 5
```
But I would say that this doesn't work as intended.

**Comment by crdoconnor** [comment]
Good catch! Thanks for pointing this out. I'll try and implement a fix tonight or tomorrow.

**Comment by crdoconnor** [comment]
Fixed in version 0.11.5. Thanks again @karantan.

**Comment by karantan** [comment]
Wau that was fast. Kudos.
